### PR TITLE
Add controls to delete listens

### PR DESCRIPTION
**Pull request by shivam-kapila**
Add a dropdown containing an option to delete listen

### CODE REVIEW

File: listenbrainz/webserver/views/api.py [hunk 1]
```diff
@@ -348,6 +349,45 @@ def validate_token():
         })
 
 
+@api_bp.route('/delete-listen', methods=['POST'])
+@crossdomain(headers="Authorization, Content-Type")
+@ratelimit()
+def delete_listen():
+    """ 
+    Delete a particular listen from the currently logged-in user's listen history.
```
Comment (paramsingh) [inline listenbrainz/webserver/views/api.py:357]: Will this go in the public docs? We should make it better if so.

File: listenbrainz/webserver/static/js/src/RecentListens.tsx [hunk 2]
```diff
@@ -516,6 +516,14 @@ export default class RecentListens extends React.Component<
     return recordingMsid ? _.get(recordingFeedbackMap, recordingMsid, 0) : 0;
   };
 
+  updateListenList = (listen: Listen) => {
+    const { listens } = this.state;
```
Comment (paramsingh) [inline listenbrainz/webserver/static/js/src/RecentListens.tsx:520]: This function name could be more descriptive. Update how?

`removeListenFromListenList`?

File: listenbrainz/webserver/static/js/src/RecentListens2.test.tsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
+import { enableFetchMocks } from "jest-fetch-mock";
```
Comment (paramsingh) [inline listenbrainz/webserver/static/js/src/RecentListens2.test.tsx:1]: RecentListens2 is not a descriptive name. Maybe `RecentListensControls.test.tsx`? Why did we add a second file though? Could we not put these tests in a Seperate describe block in the old file?

**Review by paramsingh — CHANGES_REQUESTED** [review]
Would like to see screenshots and a gif of how deletion looks to the user

File: listenbrainz/webserver/views/api.py (hunk 1 above)
Comment (paramsingh) [reply listenbrainz/webserver/views/api.py:357]: Add what parameters it takes, what it returns etc.

Comment (shivam-kapila) [reply listenbrainz/webserver/views/api.py:357]: done

File: listenbrainz/webserver/static/js/src/RecentListens.tsx (hunk 2 above)
Comment (shivam-kapila) [reply listenbrainz/webserver/static/js/src/RecentListens.tsx:520]: done

File: listenbrainz/webserver/static/js/src/RecentListens2.test.tsx (hunk 3 above)
Comment (shivam-kapila) [reply listenbrainz/webserver/static/js/src/RecentListens2.test.tsx]: done

File: listenbrainz/webserver/views/api.py [hunk 4]
```diff
@@ -348,6 +349,58 @@ def validate_token():
...
```
Comment (paramsingh) [inline listenbrainz/webserver/views/api.py:357]:
```suggestion
    Delete a particular listen from the a user's listen history.
```

This'll delete listens from whichever auth token it gets

**Review by paramsingh — APPROVED** [review]
lgtm, will merge when build passes.

**Comment by paramsingh** [comment]
@brainzbot retest this please

**Pull request merged**